create/cloudvm.go

```diff
@@ -0,0 +1,137 @@
+	PublicKeysFromFile  string            `short:"cloud-config-f" predictor:"file" help:"CloudConfig via file. Has precedents over args. PublicKeys specifies the SSH Public Keys that can be used to connect to the VM as root. The keys are expected to be in SSH format as defined in RFC4253. Immutable after creation."`
+	CloudConfig         string            `default:"" help:"CloudConfig allows to pass custom cloud config data (https://cloudinit.readthedocs.io/en/latest/topics/format.html#cloud-config-data) to the cloud VM. If a CloudConfig is passed, the PublicKey parameter is ignored. Immutable after creation."`
+	CloudConfigFromFile string            `short:"cloud-config-f" predictor:"file" help:"CloudConfig via file. Has precedents over args. CloudConfig allows to pass custom cloud config data (https://cloudinit.readthedocs.io/en/latest/topics/format.html#cloud-config-data) to the cloud VM. If a CloudConfig is passed, the PublicKey parameter is ignored. Immutable after creation."`
+	Wait                bool              `default:"true" help:"Wait until CloudVM is created."`
+	WaitTimeout         time.Duration     `default:"600s" help:"Duration to wait for CloudVM getting ready. Only relevant if --wait is set."`
+}
+
+func (cmd *cloudVMCmd) Run(ctx context.Context, client *api.Client) error {
+	cloudVM, err := cmd.newCloudVM(client.Project)
+	if err != nil {
+		return err
+	}
+
+	c := newCreator(client, cloudVM, "cloudvirtualmachine")
+	ctx, cancel := context.WithTimeout(ctx, cmd.WaitTimeout)
+	defer cancel()
+
+	if err := c.createResource(ctx); err != nil {
+		return err
+	}
+
+	if !cmd.Wait {
+		return nil
+	}
+
+	return c.wait(ctx, waitStage{
+		objectList: &infrastructure.CloudVirtualMachineList{},
+		onResult: func(event watch.Event) (bool, error) {
+			if c, ok := event.Object.(*infrastructure.CloudVirtualMachine); ok {
+				return isAvailable(c), nil
+			}
+			return false, nil
+		},
+	},
+	)
+}
+
+func (cmd *cloudVMCmd) newCloudVM(namespace string) (*infrastructure.CloudVirtualMachine, error) {
+	name := getName(cmd.Name)
+
+	cloudVM := &infrastructure.CloudVirtualMachine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: infrastructure.CloudVirtualMachineSpec{
+			ResourceSpec: runtimev1.ResourceSpec{
+				WriteConnectionSecretToReference: &runtimev1.SecretReference{
+					Name:      "cloudvirtualmachine-" + name,
+					Namespace: namespace,
+				},
+			},
+			ForProvider: infrastructure.CloudVirtualMachineParameters{
+				Location:    meta.LocationName(cmd.Location),
+				MachineType: infrastructure.MachineType(cmd.MachineType),
+				Hostname:    cmd.Hostname,
+				PowerState:  infrastructure.VirtualMachinePowerState(cmd.PowerState),
+				OS:          infrastructure.CloudVirtualMachineOS(infrastructure.OperatingSystem(cmd.OS)),
+				PublicKeys:  cmd.PublicKeys,
+			},
+		},
+	}
+
+	if len(cmd.PublicKeysFromFile) == 0 {
+		cloudVM.Spec.ForProvider.PublicKeys = cmd.PublicKeys
```

Comment:
Isn't this redundant? We already set it to `cmd.PublicKeys` so I guess you just need the `else` case here.

Reply:
yup you are right added it for both